blobfile/_azure.py

```diff
@@ -1024,27 +1025,33 @@ def create_page_iterator(
 
 class StreamingReadFile(BaseStreamingReadFile):
     def __init__(self, conf: Config, path: str, size: Optional[int]) -> None:
+        self._version: Optional[str] = None
         if size is None:
             st = maybe_stat(conf, path)
             if st is None:
                 raise FileNotFoundError(f"No such file or directory: '{path}'")
             size = st.size
+            self._version = st.version

```

Comment:
it's a little surprising that this would be controlled by the size arg. i kind of want to deprecate that and do this unconditionally... (looks like almost no one uses it anyway)